### DISCUSSION TITLE
Size as an immediate in `array.new_fixed`

### DISCUSSION THREAD

**Discussion by patrick96** [WebAssembly/binaryen#5768] [Q&A] [answered]
It looks like the parser used in `wasm-as` does not support providing the size of the array in the `array.new_fixed` instruction as an immediate value. Is that intentional?

From what I can tell from the [GC MVP document](https://github.com/WebAssembly/gc/blob/main/proposals/gc/MVP.md#arrays), an immediate value is required as part of the instruction. Their test suite seems to confirm this:

https://github.com/WebAssembly/gc/blob/099b55020af2ac29fb6f2886581de057d7372b02/test/core/gc/array.wast#L112-L114

I just wanted to first check whether there is a reason for this. I'm happy to open up a proper bug report if anyone thinks this should be looked into.

**Comment by kripken** [comment, marked as answer]
Yes, this looks like a bug in our text parsing and emitting. Binaryen IR is structured so the immediate is not strictly needed in theory, but we should add it to follow the spec.

**Reply by kripken** [reply]
I created https://github.com/WebAssembly/binaryen/issues/5769 to track this.
